tests/test_sim_result.py

```diff
@@ -36,6 +36,121 @@ def test_pickle(self):
         pickle.dump(result, open('model_test.pkl', 'wb'))
         result2 = pickle.load(open('model_test.pkl', 'rb'))
         self.assertEqual(result, result2)
+
+    def test_extend(self):
+        NUM_ELEMENTS = 5 # Creating two result objects
+        time = list(range(NUM_ELEMENTS))
+        state = [i * 2.5 for i in range(NUM_ELEMENTS)]
+        result = SimResult(time, state)
+        NUM_ELEMENTS = 10
+        time = list(range(NUM_ELEMENTS))
+        state = [i * 10.0 for i in range(NUM_ELEMENTS)]
+        result2 = SimResult(time, state)
+        self.assertEqual(result.times, [0, 1, 2, 3, 4])
+        self.assertEqual(result2.times, [0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+        self.assertEqual(result.data, [0.0, 2.5, 5.0, 7.5, 10.0]) # Assert data is correct before extending
+        self.assertEqual(result2.data, [0.0, 10.0, 20.0, 30.0, 40.0, 50.0, 60.0, 70.0, 80.0, 90.0])
+        
+        result.extend(result2) # Extend result with result2
+        self.assertEqual(result.times, [0, 1, 2, 3, 4, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+        self.assertEqual(result.data, [0.0, 2.5, 5.0, 7.5, 10.0, 0.0, 10.0, 20.0, 30.0, 40.0, 50.0, 60.0, 70.0, 80.0, 90.0])
+    
+    def test_index(self):
+        NUM_ELEMENTS = 5 # Creating two result objects
+        time = list(range(NUM_ELEMENTS))
+        state = [i * 2.5 for i in range(NUM_ELEMENTS)]
+        result = SimResult(time, state)
+        # other_arg = {2.5:1}
+        # other_arg = {1:2.5}
+        other_arg = 2.5
+
+        self.assertEqual(result.index(other_arg), 1)
```

Comment:
Let's test the behavior for when the other arg doesn't exist as well (exception should be raised)